metals/src/main/scala/scala/meta/metals/compiler/ScalacProvider.scala

```diff
@@ -58,12 +60,22 @@ class ScalacProvider() extends LazyLogging {
   def loadNewCompilerGlobals(
       config: CompilerConfig
   ): Effects.InstallPresentationCompiler = {
-    logger.info(s"Loading new compiler from config $config")
-    val compiler =
-      ScalacProvider.newCompiler(config.classpath, config.scalacOptions)
-    compilerByConfigOrigin(config.origin) = config -> compiler
-    config.sources.foreach { path =>
-      compilerByPath(Uri(path)) = compiler
+    if (config.scalaVersion.major != 2 || config.scalaVersion.minor != 12) {
```

Comment:
Nitpick, I would typically flip the condition to be `if (version == 2.12) <happy path> else <error path>`

Reply:
Makes sense: that pleases the CPU branch predictor ^^